frappe/core/doctype/communication/communication.py

```diff
@@ -387,13 +387,13 @@ def parse_email(communication, email_strings):
 		if email_string:
```

Comment:
- Add a check at the function start for checking if  Enable Automatic Link is enabled or not
` if not frappe.db.get_value("Email Account", {"enable_automatic_linking": 1}): return`

Reply:
I did think about adding in some type of checking if linking is enabled and checking the DocType is valid. As far as I could tell, parse_email is used by every single email whether Automatic Linking is enabled or not (I don't know why).

Unfortunately, your suggestion would not actually work until after the email_string is split apart, which is the most strenuous piece of this function. 

After the email is split, the "username" fragment could be added to the domain name and the suggested db call is made, however, the check on line 395 (below) does effectively the same thing and does not add another DB call every time the function runs. I do think your suggestion is valid, but it would need to be done further upstream and is out of scope for this particular "hotfix")

Line 395: Check to see if there is a valid reference and it should be linked to the Communication
`if doctype and docname and frappe.db.exists(doctype, docname):`

Reply:
- `parse_email` was implemented in the first place for automatic email linking feature, so technically this piece of code should only run if its enabled.
- So the check will ensure that the function is executed only if automatic linking is enabled, and the changes that you made, at line 392 will further act as fail proof

Reply:
I don't really agree with this particular approach but I do not have the bandwidth to refactor upstream where the checking should take place.

In order to move this forward, I have added the recommended check to the beginning of the function